v8/upgrade_metadata_v8/upgrade_metadata_v8.py

```diff
@@ -0,0 +1,182 @@
+                            source_text = meta_file.read()
+                        if not is_two_file:
+                            _, content_str = extractor.split_metadata_from_text(source_text)
+                        meta = extractor.extract_text(source_text)
+
+                        # Consider metadata mappings
+                        sources = {
+                            'tags': 'tags',
+                            'status': 'status',
+                            'has_math': 'has_math',
+                        }
+                        for foreign, ours in self.site.config.get('METADATA_MAPPING', {}).get(extractor.map_from, {}).items():
+                            if ours in sources:
+                                sources[ours] = foreign
+                        for meta_key, hook in self.site.config.get('METADATA_VALUE_MAPPING', {}).get(extractor.map_from, {}).items():
+                            if meta_key in sources.values():
+                                L.warn('Cannot convert {0} (language {1}): a metadata value mapping is defined for "{2}"!'.format(fname, lang, meta_key))
+
+                        # Update metadata
+                        updated = False
+                        tags = meta.get(sources['tags'], [])
+                        tags_are_string = False
+                        if not isinstance(tags, dict):
+                            tags_are_string = True
+                            tags = [tag.strip() for tag in tags.split(',') if tag.strip()]
+
+                        if 'draft' in [_.lower() for _ in tags]:
+                            tags.remove('draft')
+                            meta[sources['status']] = 'draft'
+                            updated = True
+
+                        if 'private' in tags:
+                            tags.remove('private')
+                            meta[sources['status']] = 'private'
+                            updated = True
+
+                        if 'mathjax' in tags:
+                            tags.remove('mathjax')
+                            meta[sources['has_math']] = 'yes'
+                            updated = True
+
+                        if tags_are_string:
+                            meta[sources['tags']] = ', '.join(tags)
+
+                        if not updated:
+                            # Nothing to do (but successful)!
+                            converted = True
+                            continue
+
+                        # Recombine metadata with post text if necessary, and write back to file
+                        meta_str = extractor.write_metadata(meta)
+                        print(meta_str)
+                        if not is_two_file:
+                            if extractor.name == 'nikola':
+                                final_str = meta_str + '\n\n' + content_str
+                            elif extractor.name == 'yaml':
+                                final_str = meta_str + '\n---\n' + content_str
```

Comment:
Is there a way not to hardcode this?

Reply:
We could extend the metadata extractors so that they offer a recombine function which does this. That would avoid this. If this is ok for you, I'll start a PR for that.

Reply:
Aaah, I missed that part :)

Reply:
You could fix 2\. by checking if the format is reST.